streams/src/main/java/org/apache/kafka/streams/processor/internals/metrics/ProcessorNodeMetrics.java

```diff
@@ -136,6 +154,86 @@ public static Sensor processAtSourceSensor(final String threadId,
         );
     }
 
+    public static Sensor bytesConsumedSensor(final String threadId,
+                                             final String taskId,
+                                             final String processorNodeId,
+                                             final String topic,
+                                             final StreamsMetricsImpl streamsMetrics) {
+        final String sensorName = processorNodeId + "-" + BYTES_CONSUMED;
+        final Sensor sensor =
+            streamsMetrics.nodeLevelSensor(threadId, taskId, processorNodeId, sensorName, RecordingLevel.INFO);
+        final Map<String, String> tagMap = new HashMap<>(streamsMetrics.nodeLevelTagMap(threadId, taskId, processorNodeId));
+        tagMap.put("topic", topic);
+        addSumMetricToSensor(
+            sensor,
+            PROCESSOR_NODE_LEVEL_GROUP,
+            tagMap,
+            BYTES_CONSUMED,
+            BYTES_CONSUMED_TOTAL_DESCRIPTION
+        );
+        return sensor;
+    }
+
+    public static Sensor recordsConsumedSensor(final String threadId,
+                                               final String taskId,
+                                               final String processorNodeId,
+                                               final String topic,
+                                               final StreamsMetricsImpl streamsMetrics) {
+        final String sensorName = processorNodeId + "-" + RECORDS_CONSUMED;
+        final Sensor sensor =
+            streamsMetrics.nodeLevelSensor(threadId, taskId, processorNodeId, sensorName, RecordingLevel.INFO);
+        final Map<String, String> tagMap = new HashMap<>(streamsMetrics.nodeLevelTagMap(threadId, taskId, processorNodeId));
+        tagMap.put("topic", topic);
+        addSumMetricToSensor(
```

Comment:
Since we always record `1` for this sensor, I think we can combine this and the bytes metrics in a single sensor with `sum` and `count` (see my other comment above). Ditto for produced.